### PR TITLE
Add support for Fashion-MNIST dataset

### PR DESCRIPTION
**Pull request by AutuanLiu**
* Add support for Fashion-MNIST dataset #399 
* Add related tests in  [tests/test_mnist_simple.py](https://github.com/tensorlayer/tensorlayer/blob/master/tests/test_mnist_simple.py)

### CODE REVIEW

**Comment by luomai** [comment]
Hi @AutuanLiu thanks a lot for your contribution! nice work. Instead of adding a hint to the mnist_simple.py, could you create a new example called tutorial_fashion_mnist.py? 

cc @zsdonghao @lgarithm

File: tensorlayer/files.py [hunk 1]
```diff
@@ -117,6 +117,73 @@ def load_mnist_labels(path, filename):
...
+    path : str
+        The path that the data is downloaded to, defaults is ``data/fashionMnist/``.
+
+    Examples
+    --------
+    >>> X_train, y_train, X_val, y_val, X_test, y_test = tl.files.load_fashion_mnist_dataset(shape=(-1,784))
+    >>> X_train, y_train, X_val, y_val, X_test, y_test = tl.files.load_fashion_mnist_dataset(shape=(-1, 28, 28, 1))
+
+    """
+    path = os.path.join(path, 'fashionMnist')
```
Comment (luomai) [inline tensorlayer/files.py]: better to spell as "fashion_mnist"

**Comment by AutuanLiu** [comment]
The size, shape, format, and training/val/test set partition of Fashion-MNIST is exactly the same as the original MNIST. One can use it to test your machine learning and deep learning algorithm performance directly, and only need to change is the data set you want input. So adding tutorial_fashion_mnist.py is not necessary. (personal opinion) 
@luomai

**Comment by lgarithm** [comment]
Yes, I don't think we need yet another tutorial.

Since the fashion mnist is designed as a drop in replacement for mnist,
Can we slightly refactor the original `load_mnist_dataset` function to make it work for both datasets?

A possible solution would be 

1. change the original `load_mnist_dataset` to 
```python
def _load_mnist_dataset(shape=(-1, 784), name='mnist', path='data', url='...'):
      """A generic function to load mnist-like dataset.
....
      """
```

2. add two new functions:
```python
def load_mnist_dataset(shape=(-1, 784), ...):
      """Load the original mnist."""
      return _load_mnist_dataset(shape, name='mnist', ...)

def load_fashion_mnist_dataset(shape=(-1, 784), ...):
      """Load the fashion mnist."""
      return _load_mnist_dataset(shape, name='fashion_mnist', ...)

```

**Comment by AutuanLiu** [comment]
How about add a parameter that indicates whether it is fashion_mnist,

* load_mnist_dataset(…, flag=True)
    * flag=True for MNIST dataset
    * flag=False for fashion_MNIST dataset
@lgarithm

**Comment by lgarithm** [comment]
People may create a new kind of mnist in the future.
If that happens, the bool flag would not be enough for extension.
I think url (for download) and name (for logging) should be generic enough for other mnist-like dataset, and pass them in could avoid using `if` `else` on the `flag` variable in the implementation of `load_mnist_dataset`.

**Comment by AutuanLiu** [comment]
OK

**Comment by lgarithm** [comment]
‘’’
1 files need to be formatted, run the following commands to fix
yapf -i tensorlayer/files.py
‘’’

**Comment by luomai** [comment]
LGTM @AutuanLiu Thanks a lot for addressing our comments. Merging it now.

**Pull request merged**